vsphere_server_plugin/server.py

```diff
@@ -315,6 +315,124 @@ def stop(ctx, server_client, server, os_family):
     ctx.logger.info('Succeessfully stop server {name}'.format(name=vm_name))
 
 
+@op
+@with_server_client
+def freeze_suspend(ctx, server_client, server, os_family):
+    if ctx.instance.runtime_properties.get('use_existing_resource'):
+        ctx.logger.info('Used existing resource.')
+        return
+    server_obj = get_server_by_context(ctx, server_client, server, os_family)
+    if server_obj is None:
+        raise NonRecoverableError(
+            "Cannot suspend server - server doesn't exist for node: {0}"
+            .format(ctx.instance.id))
+    vm_name = get_vm_name(ctx, server, os_family)
+    ctx.logger.info('Preparing to suspend server {name}'.format(name=vm_name))
+    server_client.suspend_server(server_obj)
+    ctx.logger.info('Succeessfully suspend server {name}'.format(name=vm_name))
+
+
+@op
+@with_server_client
+def freeze_resume(ctx, server_client, server, os_family):
+    if ctx.instance.runtime_properties.get('use_existing_resource'):
+        ctx.logger.info('Used existing resource.')
+        return
+    server_obj = get_server_by_context(ctx, server_client, server, os_family)
+    if server_obj is None:
+        raise NonRecoverableError(
+            "Cannot resume server - server doesn't exist for node: {0}"
+            .format(ctx.instance.id))
+    vm_name = get_vm_name(ctx, server, os_family)
+    ctx.logger.info('Preparing to resume server {name}'.format(name=vm_name))
+    server_client.start_server(server_obj)
+    ctx.logger.info('Succeessfully resume server {name}'.format(name=vm_name))
+
+
+@op
+@with_server_client
+def snapshot_create(ctx, server_client, server, os_family, snapshot_name,
+                    snapshot_incremental):
+    if ctx.instance.runtime_properties.get('use_existing_resource'):
+        ctx.logger.info('Used existing resource.')
+        return
+    if not snapshot_name:
+        raise NonRecoverableError(
+            'Backup name must be provided.'
+        )
+    if not snapshot_incremental:
+        ctx.logger.info("Create backup for VM is unsupported.")
```

Comment:
What do we mean by this? If every snapshot has to be incremental why is it even an option?

Reply:
Because we have 2 kind of backups in utilities: 
* incremental snapshots - directly supported by vsphere
* backup - something that will create "image" of instance. 

We need to have 2 options - because user can have different plugins in same blueprint - and if we raise exception - user will receive system in something at the middle.